Error: node_modules/antd/lib/tree-select/index.d.ts(18,18) | Error: node_modules/antd/lib/tree/Tree.d.ts(86,18)

### Reproduction link

[https://github.com/maifeeulasad/react-pipeline/tree/8521b5638e3634d0dadb63196acbb31abc3fa165](https://github.com/maifeeulasad/react-pipeline/tree/8521b5638e3634d0dadb63196acbb31abc3fa165)

### Steps to reproduce

```
clone
yarn
yarn build
```

### What is expected?

A proper build

### What is actually happening?

Getting the following error:
```
Run yarn build
  yarn build
  shell: /usr/bin/bash -e {0}
yarn run v1.22.19
$ tsc && vite build
Error: node_modules/antd/lib/tree-select/index.d.ts(18,18): error TS2430: Interface 'TreeSelectProps<ValueType, OptionType>' incorrectly extends interface 'Omit<TreeSelectProps<ValueType, OptionType>, "mode" | "getInputElement" | "inputIcon" | "backfill" | "showTreeIcon" | "treeMotion" | "treeLine">'.
  Types of property 'switcherIcon' are incompatible.
    Type 'SwitcherIcon' is not assignable to type 'IconType'.
      Type '(props: { expanded: boolean; }) => ReactNode' is not assignable to type 'IconType'.
        Type '(props: { expanded: boolean; }) => ReactNode' is not assignable to type '(props: TreeNodeProps<DataNode>) => ReactNode'.
          Types of parameters 'props' and 'props' are incompatible.
            Type 'TreeNodeProps<DataNode>' is not assignable to type '{ expanded: boolean; }'.
              Types of property 'expanded' are incompatible.
                Type 'boolean | undefined' is not assignable to type 'boolean'.
                  Type 'undefined' is not assignable to type 'boolean'.
Error: node_modules/antd/lib/tree/Tree.d.ts(86,18): error TS2430: Interface 'TreeProps<T>' incorrectly extends interface 'Omit<TreeProps<T>, "direction" | "draggable" | "prefixCls" | "showLine">'.
  Types of property 'icon' are incompatible.
    Type 'ReactNode | ((nodeProps: AntdTreeNodeAttribute) => ReactNode)' is not assignable to type 'IconType'.
      Type '(nodeProps: AntdTreeNodeAttribute) => ReactNode' is not assignable to type 'IconType'.
        Type '(nodeProps: AntdTreeNodeAttribute) => ReactNode' is not assignable to type '(props: TreeNodeProps<DataNode>) => ReactNode'.
          Types of parameters 'nodeProps' and 'props' are incompatible.
            Type 'TreeNodeProps<DataNode>' is not assignable to type 'AntdTreeNodeAttribute'.
              Types of property 'eventKey' are incompatible.
                Type 'Key | undefined' is not assignable to type 'string'.
                  Type 'undefined' is not assignable to type 'string'.
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: Process completed with exit code 2.
```

| Environment | Info |
| --- | --- |
| antd | 4.21.3 |
| React | 18.1.0 |
| System |  ubuntu-latest |
| Browser | node 14 |

---
https://github.com/maifeeulasad/react-pipeline/runs/6949098925?check_suite_focus=true

<!-- generated by ant-design-issue-helper. DO NOT REMOVE -->

--- a/components/tree-select/index.tsx
+++ b/components/tree-select/index.tsx
@@ -44,6 +44,7 @@ export interface TreeSelectProps<
     | 'getInputElement'
     | 'backfill'
     | 'treeLine'
+    | 'switcherIcon'
   > {
   suffixIcon?: React.ReactNode;
   size?: SizeType;
@@ -52,7 +53,7 @@ export interface TreeSelectProps<
   bordered?: boolean;
   treeLine?: TreeProps['showLine'];
   status?: InputStatus;
-  switcherIcon?: SwitcherIcon;
+  switcherIcon?: SwitcherIcon | RcTreeSelectProps<ValueType, OptionType>['switcherIcon'];
 }
 
 const InternalTreeSelect = <OptionType extends BaseOptionType | DefaultOptionType = BaseOptionType>(

--- a/components/tree/Tree.tsx
+++ b/components/tree/Tree.tsx
@@ -102,7 +102,10 @@ interface DraggableConfig {
 }
 
 export interface TreeProps<T extends BasicDataNode = DataNode>
-  extends Omit<RcTreeProps<T>, 'prefixCls' | 'showLine' | 'direction' | 'draggable' | 'icon'> {
+  extends Omit<
+    RcTreeProps<T>,
+    'prefixCls' | 'showLine' | 'direction' | 'draggable' | 'icon' | 'switcherIcon'
+  > {
   showLine?: boolean | { showLeafIcon: boolean };
   className?: string;
   /** 是否支持多选 */
@@ -143,7 +146,7 @@ export interface TreeProps<T extends BasicDataNode = DataNode>
     | ((nodeProps: AntdTreeNodeAttribute) => React.ReactNode)
     | React.ReactNode
     | RcTreeProps<T>['icon'];
-  switcherIcon?: SwitcherIcon;
+  switcherIcon?: SwitcherIcon | RcTreeProps<T>['switcherIcon'];
   prefixCls?: string;
   children?: React.ReactNode;
   blockNode?: boolean;

--- a/components/tree/Tree.tsx
+++ b/components/tree/Tree.tsx
@@ -102,7 +102,7 @@ interface DraggableConfig {
 }
 
 export interface TreeProps<T extends BasicDataNode = DataNode>
-  extends Omit<RcTreeProps<T>, 'prefixCls' | 'showLine' | 'direction' | 'draggable'> {
+  extends Omit<RcTreeProps<T>, 'prefixCls' | 'showLine' | 'direction' | 'draggable' | 'icon'> {
   showLine?: boolean | { showLeafIcon: boolean };
   className?: string;
   /** 是否支持多选 */
@@ -139,7 +139,10 @@ export interface TreeProps<T extends BasicDataNode = DataNode>
   draggable?: DraggableFn | boolean | DraggableConfig;
   style?: React.CSSProperties;
   showIcon?: boolean;
-  icon?: ((nodeProps: AntdTreeNodeAttribute) => React.ReactNode) | React.ReactNode;
+  icon?:
+    | ((nodeProps: AntdTreeNodeAttribute) => React.ReactNode)
+    | React.ReactNode
+    | RcTreeProps<T>['icon'];
   switcherIcon?: SwitcherIcon;
   prefixCls?: string;
   children?: React.ReactNode;

